Porting/valgrindpp.pl

```diff
@@ -197,7 +198,7 @@ sub filter {
 
   # Get all the valgrind output lines
   my @l = do {
-    my $fh = new IO::File $_ or die "$0: cannot open $_ ($!)\n";
+    my $fh = IO::File->new($_) or die "$0: cannot open $_ ($!)\n";
```

Comment:
Should probably be `IO::File->new($_, 'r')`.

Reply:
The second argument does not default to `r`. The linked code shows that it acts like 2-arg open, which tries to parse an open mode and a filename from the second argument. That's not the same as always opening the file for reading. For correctness, we never want to reinterpret filenames; i.e. always use 3-arg open or the corresponding `IO::File->new($file, $mode)` form.

Reply:
Thanks @make, will amend that

Reply:
Force-pushed